### ISSUE TITLE
Improve custom theme support in modals/popups/custom components

### ISSUE THREAD

**Issue by sharon-wang** [posit-dev/positron#3799]
## Background

Users can select additional themes beyond the regular and high contrast light and dark themes. The "Tomorrow Night Blue" theme is built-in, but is lacking a bit in colour contrast in our custom modals/popups. When using other custom themes, sometimes the UI is not even legible.

Another example: https://github.com/posit-dev/positron/issues/3735

#### Delete All Variables
<img width="383" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/ecde41b4-4457-45c5-aeec-4b41974536ca">

#### Project Wizard
<img width="709" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/cf20f222-03e3-44a9-8d7e-c936c90099ee">

<img width="709" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/ac2f1cc1-098d-4e8c-a6aa-33282bc3d7ca">

#### New Folder

<img width="412" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/d1300e94-8a30-4a76-8b3b-655f7a1d7bdc">

#### New Folder from Git

<img width="414" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/c85b00ee-e659-4110-acd0-51a4030a72ab">

#### Data Explorer - column menu

<img width="379" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/08bbf59a-3247-4773-a78a-ddab8bc67c3e">

#### Data Explorer - add filter

<img width="290" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/aab4b5ad-aa53-44ed-a870-7dc837929951">

#### Project Wizard

<img width="706" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/9c22204f-a8dd-4ae9-b6a3-5875674080bd">

#### New Folder

<img width="409" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/f5a39b54-78d4-445d-9e11-813e8b93769d">

#### Delete All Variables

<img width="382" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/e4129f00-0fcb-4434-8173-bb77e8f3644a">

#### Data Explorer - add filter

<img width="285" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/97a73f89-0369-4700-8aba-74ce5fc6f3bd">

#### Project Wizard

<img width="708" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/4bc39f21-4fde-45e0-a898-a362520d5d1b">

#### New Folder

<img width="409" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/179b5129-e04d-4bc0-a73c-bec386b3042a">

#### Delete All Variables

<img width="382" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/b13f176b-45e7-4fd8-9393-2b70354dcd44">

#### Data Explorer - add filter

<img width="285" alt="image" src="https://github.com/posit-dev/positron/assets/25834218/c8671184-61e4-497e-a27c-de529a77dabb">

## Thoughts on Solution

Although we may not be able to ensure that every custom theme will be reasonably visually accessible, I think we should be able to set some sensible fallback colours which some of these themes may rely on. Perhaps we can even override the custom theme at times, to ensure that text is legible?

**Comment by juliasilge** [comment]
When we investigate what is going on here, let's also look at problems of the category described in https://github.com/posit-dev/positron/discussions/7319:

<img width="1601" alt="Screenshot 2025-04-22 at 08 45 20" src="https://github.com/user-attachments/assets/2777d44f-d23a-451f-b1f5-8badc6ab7010" />

**Comment by petetronic** [comment]
Sending back to Triage, is this a must have for 1.0? Can we do this in Post-RC?

**Comment by juliasilge** [comment]
We think this is resolved enough for now, and will look for any future user reports.
